explorer/src/components/common/VerifiedBadge.tsx

```diff
@@ -0,0 +1,91 @@
+import { PublicKey } from "@solana/web3.js";
+import { useEffect, useState } from "react";
+
+export type AnchorBuild = {
+  aborted: boolean;
+  address: string;
+  created_at: string;
+  updated_at: string;
+  descriptor: string[];
+  docker: string;
+  id: number;
+  name: string;
+  sha256: string;
+  upgrade_authority: string;
+  verified: string;
+  verified_slot: number;
+  state: string;
+};
+
+/**
+ * Returns a verified build from the anchor registry. null if no such
+ * verified build exists, e.g., if the program has been upgraded since the
+ * last verified build.
+ */
+async function getVerifiedBuild(
+  deploySlot: number,
+  programId: PublicKey,
+  limit: number = 5
+) {
+  const url = `https://anchor.projectserum.com/api/v0/program/${programId.toString()}/latest?limit=${limit}`;
+  const latestBuildsResp = await fetch(url);
+
+  // Filter out all non successful builds.
+  const latestBuilds = (await latestBuildsResp.json()).filter(
+    (b: AnchorBuild) =>
+      !b.aborted && b.state === "Built" && b.verified === "Verified"
+  ) as AnchorBuild[];
+
+  if (latestBuilds.length === 0) {
+    return null;
+  }
+
+  // Get the latest build.
+  const build = latestBuilds[0];
+
+  // Has the program been upgraded since the last build?
+  if (deploySlot !== build.verified_slot) {
+    return null;
+  }
+
+  // Success.
+  return build;
+}
+
+export function VerifiedBadge({
+  programAddress,
+  deploySlot,
+}: {
+  programAddress: PublicKey;
+  deploySlot: number;
+}) {
+  const [loading, setLoading] = useState(true);
+  const [verifiedBuild, setVerifiedBuild] = useState<AnchorBuild | null>();
+
+  useEffect(() => {
+    setLoading(true);
+    getVerifiedBuild(deploySlot, programAddress).then((build) => {
+      setVerifiedBuild(build);
+      setLoading(false);
+    });
+  }, [deploySlot, programAddress, setVerifiedBuild, setLoading]);
+
+  if (loading) return <span className="ms-2 badge bg-dark rank">checking</span>;
+  if (verifiedBuild) {
+    return (
+      <span
```

Comment:
Can you make this an anchor element with a dynamic href? Then you don't need to do the window open call manually.